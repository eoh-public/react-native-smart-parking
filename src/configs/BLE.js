export default {
  BLE_REMOTE_SERVICE_UUID: '4fafc201-1fb5-459e-8fcc-c5c9c331914b',
  BLE_REMOTE_CHARACTERISTIC_UUID: 'beb5483e-36e1-4688-b7f5-ea07361b26a8',
};
