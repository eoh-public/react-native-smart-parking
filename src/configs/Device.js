/** @format */

import { Dimensions, PixelRatio, Platform } from 'react-native';

const { width, height } = Dimensions.get('window');

const isIphoneX =
  Platform.OS === 'ios' &&
  !Platform.isPad &&
  !Platform.isTVOS &&
  (height >= 812 || width >= 812);

export default {
  pixelRatio: PixelRatio.get(),
  screenWidth: width,
  screenHeight: height,
  isIOS: Platform.OS === 'ios',
  isIphoneX,
  ToolbarHeight: isIphoneX ? 44 : Platform.OS === 'ios' ? 20 : 0,
  TopbarHeight: isIphoneX ? 88 : Platform.OS === 'ios' ? 64 : 44, //incule toolbar
  androidSmallHeight: Platform.OS === 'android' && height < 600,
};
