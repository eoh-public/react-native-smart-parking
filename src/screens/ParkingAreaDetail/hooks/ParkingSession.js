import { useCallback, useState } from 'react';
import { API } from '../../../configs';
import { axiosGet } from '../../../utils/Apis/axios';
import moment from 'moment';

const useParkingSession = (id, preBook, numBookHour) => {
  const [parkingSessionData, setParkingSessionData] = useState([]);
  const [bookTime, setBookTime] = useState({
    arriveAt: moment(),
    numBookHour: numBookHour || 1,
  });

  const getParkingSession = useCallback(async () => {
    if (preBook) {
      const { data, success } = await axiosGet(
        API.PARKING.AVAILABLE_TIME_SLOTS(id)
      );
      if (success) {
        const datetimeData = data.map((item) => ({
          ...item,
          time: moment(item.time),
        }));

        setParkingSessionData(datetimeData);
        if (datetimeData.length) {
          const initTime = datetimeData[0].time;
          setBookTime({ arriveAt: initTime, numBookHour: 1 });
        }
      }
    } else {
      setBookTime({ numBookHour: 1, arriveAt: moment() });
    }
  }, [id, preBook]);

  return {
    bookTime,
    setBookTime,
    parkingSessionData,
    getParkingSession,
  };
};

export { useParkingSession };
