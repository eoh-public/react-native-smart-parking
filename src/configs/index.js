import { SPConfig, initSPConfig } from './SPConfig';
import API from './API';
import { Colors } from './Colors';
import { Constants } from './Constants';
import Theme from './Theme';
import Images from './Images';
import Styles from './Styles';
import BLE from './BLE';
import Device from './Device';
import { Fonts } from './Fonts';
export {
  API,
  BLE,
  Colors,
  Constants,
  Theme,
  Styles,
  Images,
  Device,
  Fonts,
  SPConfig,
  initSPConfig,
};
