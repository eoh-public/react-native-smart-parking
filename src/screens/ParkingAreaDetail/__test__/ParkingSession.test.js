import React from 'react';
import { create, act } from 'react-test-renderer';
import ParkingSession from '../compenents/ParkingSession';
import moment from 'moment';
import { Text, TouchableOpacity } from 'react-native';
import { TESTID } from '../../../configs/Constants';
import { Colors } from '../../../configs';

describe('Test ParkingSession', () => {
  let data;
  const mockSetBookTime = jest.fn();

  beforeEach(() => {
    Date.now = jest.fn(() => new Date('2021-01-24T12:00:00.000Z'));
    data = {
      bookTime: {
        arriveAt: moment(),
        numBookHour: 1,
      },
      setBookTime: mockSetBookTime,
      parkingSessionData: [
        {
          time: moment(),
          price: 1000,
          id: 1,
        },
        {
          time: moment(),
          price: 1000,
          id: 2,
        },
      ],
      preBook: false,
      spotNumber: 'HU1',
    };
  });
  let wrapper;

  test('create', () => {
    act(() => {
      wrapper = create(<ParkingSession {...data} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('create without spotNumber', () => {
    data.spotNumber = '';
    act(() => {
      wrapper = create(<ParkingSession {...data} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('onPress ArriveItem, with preBook', () => {
    data.preBook = true;
    const styleContainer = {
      width: 80,
      borderWidth: 1,
      borderColor: '#E8E8E8',
      paddingVertical: 8,
      alignItems: 'center',
      marginRight: 16,
    };
    const styleSelected = [styleContainer, { borderColor: Colors.Primary }];
    const styleNotSelected = [styleContainer, { borderColor: Colors.Gray4 }];

    act(() => {
      wrapper = create(<ParkingSession {...data} />);
    });
    const instance = wrapper.root;

    const buttonsArriveItem = instance.findAll(
      (el) =>
        el.props.testID === TESTID.ARRIVE_ITEM && el.type === TouchableOpacity
    );

    expect(buttonsArriveItem).toHaveLength(2);
    expect(buttonsArriveItem[0].props.style).toEqual(styleSelected);
    expect(buttonsArriveItem[1].props.style).toEqual(styleNotSelected);

    act(() => {
      buttonsArriveItem[1].props.onPress();
    });

    expect(buttonsArriveItem[1].props.style).toEqual(styleSelected);
    expect(mockSetBookTime).toHaveBeenCalledWith({
      arriveAt: moment(),
      numBookHour: 1,
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('onPress plus onChangeHour, with preBook', () => {
    data.preBook = true;
    act(() => {
      wrapper = create(<ParkingSession {...data} />);
    });
    const instance = wrapper.root;

    const buttonPlus = instance.find(
      (el) => el.props.testID === TESTID.PLUS && el.type === TouchableOpacity
    );
    const textHour = instance.find(
      (el) => el.props.testID === TESTID.HOUR && el.type === Text
    );

    expect(textHour.instance.props.children).toEqual('1 giờ');

    act(() => {
      buttonPlus.props.onPress();
    });

    expect(textHour.instance.props.children).toEqual('2 giờ');

    expect(mockSetBookTime).toHaveBeenCalledWith({
      arriveAt: moment(),
      numBookHour: 2,
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
