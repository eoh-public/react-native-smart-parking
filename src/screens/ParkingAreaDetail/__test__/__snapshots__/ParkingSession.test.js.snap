// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ParkingSession create 1`] = `
Array [
  <View
    style={
      Object {
        "marginBottom": 8,
        "marginLeft": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "SFProDisplay-Semibold",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      Vị trí đỗ xe
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={false}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FAFAFA",
            "borderColor": "#E8E8E8",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#262626",
            "height": 40,
            "marginEnd": 5,
            "marginStart": 5,
            "textAlign": "center",
            "width": 40,
          }
        }
        underlineColorAndroid="transparent"
        value="H"
      />
      <TextInput
        allowFontScaling={true}
        editable={false}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FAFAFA",
            "borderColor": "#E8E8E8",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#262626",
            "height": 40,
            "marginEnd": 5,
            "marginStart": 5,
            "textAlign": "center",
            "width": 40,
          }
        }
        underlineColorAndroid="transparent"
        value="U"
      />
      <TextInput
        allowFontScaling={true}
        editable={false}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FAFAFA",
            "borderColor": "#E8E8E8",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#262626",
            "height": 40,
            "marginEnd": 5,
            "marginStart": 5,
            "textAlign": "center",
            "width": 40,
          }
        }
        underlineColorAndroid="transparent"
        value="1"
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 8,
          "marginLeft": 16,
        },
      ]
    }
  >
    Thời gian đỗ xe
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    Thời gian đỗ:
  </Text>,
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 16,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFAFA",
          "borderColor": "#E8E8E8",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": 48,
        }
      }
      testID="MINUS"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E8E8E8",
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
          "marginHorizontal": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#00979D",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 20,
              "lineHeight": 28,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
        testID="HOUR"
      >
        1 giờ
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFAFA",
          "borderColor": "#E8E8E8",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": 48,
        }
      }
      testID="PLUS"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 12,
          "lineHeight": 20,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "alignSelf": "center",
          "marginBottom": 16,
          "marginTop": 8,
        },
      ]
    }
  >
    Kết thúc 1:00 PM - 24/01/2021
  </Text>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test ParkingSession create without spotNumber 1`] = `
Array [
  <View
    style={
      Object {
        "marginBottom": 8,
        "marginLeft": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "SFProDisplay-Semibold",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      Vị trí đỗ xe
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 8,
          "marginLeft": 16,
        },
      ]
    }
  >
    Thời gian đỗ xe
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    Thời gian đỗ:
  </Text>,
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 16,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFAFA",
          "borderColor": "#E8E8E8",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": 48,
        }
      }
      testID="MINUS"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E8E8E8",
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
          "marginHorizontal": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#00979D",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 20,
              "lineHeight": 28,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
        testID="HOUR"
      >
        1 giờ
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFAFA",
          "borderColor": "#E8E8E8",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": 48,
        }
      }
      testID="PLUS"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 12,
          "lineHeight": 20,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "alignSelf": "center",
          "marginBottom": 16,
          "marginTop": 8,
        },
      ]
    }
  >
    Kết thúc 1:00 PM - 24/01/2021
  </Text>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test ParkingSession onPress ArriveItem, with preBook 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 8,
          "marginLeft": 16,
        },
      ]
    }
  >
    Thời gian đỗ xe
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    Bắt đầu
  </Text>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 16,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#E8E8E8",
          "borderWidth": 1,
          "marginRight": 16,
          "opacity": 1,
          "paddingVertical": 8,
          "width": 80,
        }
      }
      testID="ARRIVE_ITEM"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginBottom": 4,
            },
          ]
        }
      >
        12:00 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 12,
              "lineHeight": 20,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        1.000 đ
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#00979D",
          "borderWidth": 1,
          "marginRight": 16,
          "opacity": 1,
          "paddingVertical": 8,
          "width": 80,
        }
      }
      testID="ARRIVE_ITEM"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginBottom": 4,
            },
          ]
        }
      >
        12:00 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 12,
              "lineHeight": 20,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        1.000 đ
      </Text>
    </View>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    Thời gian đỗ:
  </Text>,
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 16,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFAFA",
          "borderColor": "#E8E8E8",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": 48,
        }
      }
      testID="MINUS"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E8E8E8",
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
          "marginHorizontal": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#00979D",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 20,
              "lineHeight": 28,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
        testID="HOUR"
      >
        1 giờ
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFAFA",
          "borderColor": "#E8E8E8",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": 48,
        }
      }
      testID="PLUS"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 12,
          "lineHeight": 20,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "alignSelf": "center",
          "marginBottom": 16,
          "marginTop": 8,
        },
      ]
    }
  >
    Kết thúc 1:00 PM - 24/01/2021
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 16,
        "paddingHorizontal": 16,
      }
    }
  >
    <SvgMock />
    <Text
      style={
        Array [
          Object {
            "color": "#8C8C8C",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 12,
            "lineHeight": 20,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 8,
          },
        ]
      }
    >
      Bạn chỉ có thể đặt chỗ trước tối đa 1 tiếng.
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test ParkingSession onPress plus onChangeHour, with preBook 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 8,
          "marginLeft": 16,
        },
      ]
    }
  >
    Thời gian đỗ xe
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    Bắt đầu
  </Text>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 16,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#00979D",
          "borderWidth": 1,
          "marginRight": 16,
          "opacity": 1,
          "paddingVertical": 8,
          "width": 80,
        }
      }
      testID="ARRIVE_ITEM"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginBottom": 4,
            },
          ]
        }
      >
        12:00 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 12,
              "lineHeight": 20,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        1.000 đ
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#E8E8E8",
          "borderWidth": 1,
          "marginRight": 16,
          "opacity": 1,
          "paddingVertical": 8,
          "width": 80,
        }
      }
      testID="ARRIVE_ITEM"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginBottom": 4,
            },
          ]
        }
      >
        12:00 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 12,
              "lineHeight": 20,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        1.000 đ
      </Text>
    </View>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    Thời gian đỗ:
  </Text>,
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 16,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFAFA",
          "borderColor": "#E8E8E8",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": 48,
        }
      }
      testID="MINUS"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E8E8E8",
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
          "marginHorizontal": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#00979D",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 20,
              "lineHeight": 28,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
        testID="HOUR"
      >
        2 giờ
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FAFAFA",
          "borderColor": "#E8E8E8",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "width": 48,
        }
      }
      testID="PLUS"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262626",
              "fontFamily": "antoutline",
              "fontSize": 24,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 12,
          "lineHeight": 20,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "alignSelf": "center",
          "marginBottom": 16,
          "marginTop": 8,
        },
      ]
    }
  >
    Kết thúc 2:00 PM - 24/01/2021
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 16,
        "paddingHorizontal": 16,
      }
    }
  >
    <SvgMock />
    <Text
      style={
        Array [
          Object {
            "color": "#8C8C8C",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 12,
            "lineHeight": 20,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 8,
          },
        ]
      }
    >
      Bạn chỉ có thể đặt chỗ trước tối đa 1 tiếng.
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;
